lib/event_store_http_client/events/write.rb

```diff
@@ -14,14 +14,12 @@ class Write
         dependency :settings
         dependency :client
 
-        def initialize(type, data, stream_name, version)
```

Comment:
True.  So we should be able to build a CategoryStream writer that gets instantiated with a Category (which will wind up being the prefix) and accepts an id at write-time, and then a general stream writer that can write to an arbitrary stream and accepts a stream name at write-time.